src/menus/picklist/picklist.ts

```diff
@@ -37,11 +38,35 @@ export class NglPicklist {
   @Input() open: boolean;
   @Output() openChange = new EventEmitter();
 
+  @Input() filter: string | Function;
+  @Input() filterPlaceholder = 'Filter Options';
+
+  get hasFilter() {
+    return typeof(this.filter) !== 'undefined';
+  }
+
+  get filteredData() {
+    const filter = <any>this.filter; // until proper type guards are available in typescript
+    switch (typeof(this.filter)) {
+      case 'string':
+        if (/^\s*$/.test(filter)) {
```

Comment:
When the filter is the empty string, the `toString` is called against the picklist item, so the check is needed. It will simply change form according to your previous comment.